### ISSUE TITLE
run mintty(cygwin) remotely from jenkins

### ISSUE THREAD

**Issue by alnhk** [mintty/mintty#1135]
I am trying to run "configure" and "make" commands on windows platform (cygwin), from jenkins pipeline, running using this below command :

`bat "${CYGWIN_EXECTION} /cygdrive/c/JDK_BUILD_64_32/configure_64.sh ${env.BUILD_NUMBER} ${JDK11_U_PATH_CYGWIN} ${params.snclabel} "`

and

`bat "${CYGWIN_EXECTION} /cygdrive/c/JDK_BUILD_64_32/make_64.sh ${env.BUILD_NUMBER} ${JDK11_U_PATH_CYGWIN} ${params.snclabel} "`

where : 
CYGWIN_EXECTION="C:\cygwin64\bin\mintty.exe -h start /bin/bash -l -e "

and

 configure_64.sh contains below script:

`./configure --with-freetype=bundled --with-tools-dir=/cygdrive/c/progra~2/micros~1/2017/Enterprise/VC/Tools/ --with-jtreg=/cygdrive/c/jtreg-5.1-b01/ --with-version-pre=$SNCMID_VAR --with-version-build=$BUILD_NUMBER--with-version-opt= --with-boot-jdk=/cygdrive/c/jdk-10.0.2 --with-boot-jdk-jvmargs="-Xmx4G -enableassertions"`

and make_64.sh contains below script:

`./make clean SPEC=build/windows-x86_64-normal-server-release/spec.gmk`

When triggered from jenkins, the jenkins pipeline did start to execute but did not wait for the execution to finish (simply closes the cygwin dialog box, did not wait for execution to finish).

please advise the resolution to make it to wait for the "./configure" execution to finish (before moving forward to run "./make" ) ?

In short, on windows platform , we are using cygwin to compile OPENJDK source and generate windows based JRE (custom)., hence we use "./configure" and "./make" on cygwin to generate windows based binaries.

**Comment by avih** [comment]
How is mintty related to to this? mintty is a terminal, while all you want is to run a shell script.

Would you consider running xterm (or any other terminal) on linux only to run a script? Same for cygwin.

You can use the cygwin `run` command to start a script without mintty.

**Comment by mintty** [comment]
Probably not an issue. Closing.
